llarp/router/outbound_message_handler.cpp

```diff
@@ -0,0 +1,209 @@
+#include <router/outbound_message_handler.hpp>
+
+#include <messages/link_message.hpp>
+#include <router/i_outbound_session_maker.hpp>
+
+namespace llarp
+{
+
+  bool
+  OutboundMessageHandler::QueueMessage(const RouterID &remote,
+                                       const ILinkMessage *msg,
+                                       SendStatusHandler callback)
+  {
+    std::array< byte_t, MAX_LINK_MSG_SIZE > linkmsg_buffer;
+    llarp_buffer_t buf(linkmsg_buffer);
+
+    if (!EncodeBuffer(msg, buf))
+    {
+      return false;
+    }
+
+    Message message = std::make_pair(buf.sz, callback);
+
+    memcpy(message.first.data(), buf.base, buf.sz);
+
+    if (SendIfSession(remote, message))
+    {
+      return true;
+    }
+
+    bool shouldCreateSession = false;
+    {
+      util::Lock l(&_mutex);
+
+      // create queue for <remote> if it doesn't exist, and get iterator
+      auto itr_pair = outboundMessageQueue.emplace(remote, MessageQueue());
+
+      itr_pair.first->push(std::move(message));
+
+      shouldCreateSession = itr_pair->second;
+    }
+
+    if (shouldCreateSession)
+    {
+      QueueSessionCreation(remote);
+    }
+
+    return true;
+  }
+
+  //TODO: this
+  util::StatusObject
+  OutboundMessageHandler::ExtractStatus() const
+  {
+    util::StatusObject status;
+    return status;
```

Comment:
`return {};` instead?

Reply:
eh, this is fine for now.  I can change it if it really matters but since the function is still a TODO and the compiler should do the same here regardless it's probably fine.

Reply:
Not changed, can still do if necessary.